packages/context-menu/src/vaadin-context-menu-overlay.js

```diff
@@ -101,13 +102,32 @@ class ContextMenuOverlay extends OverlayElement {
     return {
       xMax: overlayRect.right - contentRect.width,
       xMin: overlayRect.left + contentRect.width,
-      yMax,
-      left: overlayRect.left,
-      right: overlayRect.right,
-      top: overlayRect.top,
-      width: contentRect.width
+      yMax
     };
   }
+
+  _updatePosition() {
+    this.style.transform = '';
+    super._updatePosition();
+
+    const isFullScreen = this.offsetWidth === window.innerWidth;
+    if (!isFullScreen && this.positionTarget && this.parentOverlay) {
+      // This overlay is positioned by a parent menu item,
+      // adjust the position by the overlay content paddings
+      const content = this.$.content;
+      const style = getComputedStyle(content);
+
+      const offsetY = this.hasAttribute('bottom-aligned')
+        ? parseFloat(style.paddingBottom)
+        : -parseFloat(style.paddingTop);
+
+      const isLeftAligned =
+        (this.hasAttribute('start-aligned') && !this.__isRTL) || (this.hasAttribute('end-aligned') && this.__isRTL);
+      const offsetX = isLeftAligned ? parseFloat(style.paddingLeft) : -parseFloat(style.paddingRight);
+
+      this.style.transform = `translate(${offsetX}px, ${offsetY}px)`;
```

Comment:
So the position mixin sets left, right, top, or bottom, and then this thing adds a transform on top of that? 😬

I'm not sure if that is actually an improvement. It seems we rather want to hook into the positioning logic of the mixin here.

Reply:
Good point. Changed the logic to adjust the same style properties (left/right/bottom/top) that are changed by the position mixin. This also eliminates the need to check for fullscreen.

Reply:
OK, that seems reasonable.